Scripts/Weapon/DashSkill.cs

```diff
@@ -0,0 +1,245 @@
+    }
+
+    private Enemy GetTargetEnemy()
+    {
+        Array bodies = targetArea.GetOverlappingBodies();
+        Enemy closestEnemy = null;
+        float lowestDegreesDifference = 0;
+
+        Vector3 forward = -GlobalTransform.basis.z;
+        foreach (PhysicsBody body in bodies)
+        {
+            Vector3 bodyDisplacement = body.GlobalTransform.origin - GlobalTransform.origin;
+            Plane upFacingPlane = new Plane(GlobalTransform.basis.y, 0);
+            Vector3 flattenedBodyDisplacement = upFacingPlane.Project(bodyDisplacement);
+
+            float degreesDifference = Mathf.Rad2Deg(forward.AngleTo(flattenedBodyDisplacement));
+
+            if (!(body is Enemy))
+                continue;
+
+            if (degreesDifference > spreadDegrees / 2)
+                continue;
+
+            if (!(body as Enemy).HasStatus<MarkStatus>())
+                continue;
+
+            if (closestEnemy == null || degreesDifference < lowestDegreesDifference)
+            {
+                closestEnemy = body as Enemy;
+                lowestDegreesDifference = degreesDifference;
+            }
+        }
+
+        // If no enemies found within main cone area, look at the sides
+        if (closestEnemy == null)
+        {
+            Array sideBodies = coneSides.GetOverlappingBodies();
+            foreach (PhysicsBody body in sideBodies)
+            {
+                if (!(body is Enemy))
+                    continue;
+
+                if (!(body as Enemy).HasStatus<MarkStatus>())
+                    continue;
+                Vector3 bodyDisplacement = body.GlobalTransform.origin - GlobalTransform.origin;
+                Plane upFacingPlane = new Plane(GlobalTransform.basis.y, 0);
+                Vector3 flattenedBodyDisplacement = upFacingPlane.Project(bodyDisplacement);
+
+                float degreesDifference = Mathf.Rad2Deg(forward.AngleTo(flattenedBodyDisplacement));
+                if (closestEnemy == null || degreesDifference < lowestDegreesDifference)
+                {
+                    closestEnemy = body as Enemy;
+                    lowestDegreesDifference = degreesDifference;
+                }
+            }
+        }
+
+        return closestEnemy;
+    }
+
+    protected override void OnAttackButtonPressed()
+    {
+        showTargetHint = true;
+    }
+
+    protected override void OnAttackButtonReleased()
+    {
+        showTargetHint = false;
+        if (isRunning)
+            return;
+        Vector3 currentLocation = GlobalTransform.origin;
+        Vector3 targetLocation;
+        if (targetEnemy == null)
+        {
+            targetLocation = currentLocation + range * (-GlobalTransform.basis.z);
+        }
+        else
+        {
+            Vector3 enemyLocation = targetEnemy.GlobalTransform.origin;
+            Dictionary ray = GetWorld().DirectSpaceState.IntersectRay(currentLocation, enemyLocation, collisionMask: 4);
+            Vector3 enemyBoundary = (Vector3)ray["position"];

```

Comment:
Need to check if `"position"` exists. I got this error out of a sudden. Not sure how to reproduce.

![image](https://user-images.githubusercontent.com/8130120/80908440-6506ca80-8d52-11ea-9729-0dbbba9b8725.png)

Reply:
technically this is not supposed to happen, but sometimes it does idk why